CFC_WebApp/api/cfc_webapp.py

```diff
@@ -352,6 +354,10 @@ def javascriptCallback(clientName, method):
   client.callJavascriptCallback(client_key, method, request.params)
   return {'status': 'ok'}
 
+@get('/fbregister')
```

Comment:
fbregister actually does call the function in socialmedia client to register the user. This facebookLogin() function may be found in socialmedia.py. It renders the template and passes in the client key. Before you pointed out that Client has an update function, I was using the manual update script to populate the client db. Anyway, the subsequent call on successful login redirects the user to a page which calls ‘getFacebookUser.get_user_from_cookies(cookies)’, which then calls the FacebookAPI to get the user’s e-mail. Of course, this could all be simplified by one call if we specify the Facebook UID or e-mail as the foreign key to the uuid_db().  

On a general note: Yes, the current implementation is a rather crude, inefficient, and opinionated one, but was meant to get the conversation going, figure out the moving parts of your system, throw out some ideas, and get folks towards consensus on what a good implementation would look like. I don’t have a full vision of how this should integrate, but the QA/QC is definitely appreciated.

Sid Feygin
sid.feygin@gmail.com



> On Jun 2, 2015, at 4:57 PM, shankari <notifications@github.com> wrote:
> 
> In CFC_WebApp/api/cfc_webapp.py <https://github.com/e-mission/e-mission-server/pull/133#discussion_r31583392>:
> 
> > @@ -352,6 +354,10 @@ def javascriptCallback(clientName, method):
> >    client.callJavascriptCallback(client_key, method, request.params)
> >    return {'status': 'ok'}
> >  
> > +@get('/fbregister')
> This is not really used anywhere in the current implementation.
> Remove until it is actually needed.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/e-mission/e-mission-server/pull/133/files#r31583392>.
>